### PR TITLE
fix(gatsby): enable static queries in `wrapRootElement`

### PR DESCRIPTION
**Pull request by pieh**
This is partial fix to https://github.com/gatsbyjs/gatsby/issues/7747

See [comment](https://github.com/gatsbyjs/gatsby/issues/7747#issuecomment-464790116) listing things to fix - this PR handles first bullet point:

It move StaticQueryContext to absolute root of components tree, making sure that everything is below that and have access to that context.

This was developed during pairing session with @oorestisime - Thanks! 

We still need to clean this up a bit: I think instead of 2 json-store files, we can have just one that exports 2 components. For sure we have to remove the commented out code.

Potentially good idea would be to add some tests for this in our e2e tests as well

### CODE REVIEW

**Comment by oorestisime** [comment]
I cleaned up a bit the code and tried to add an e2e test. not sure how this can get handled since it needs the version of gatsby we are trying to modify

i also get a weird error when adding the modified version with gatsby dev. 

```
Unhandled Rejection (TypeError): plugin.plugin[api] is not a function
./.cache/api-runner-browser.js/</exports.apiRunner/results<
/home/grok/personal/gatsby/e2e-tests/development-runtime/.cache/api-runner-browser.js:35
```

File: packages/gatsby/cache-dir/root.js [hunk 1]
```diff
@@ -9,7 +9,10 @@ import {
 } from "./navigation"
 import { apiRunner } from "./api-runner-browser"
 import loader from "./loader"
-import JSONStore from "./json-store"
+import {
+  JSONStoreForPageQueries,
+  JSONStoreForStaticQueries,
```
Comment (m-allanson) [inline packages/gatsby/cache-dir/root.js]: Maybe these could be PageQueryStore and StaticQueryStore (or similar)? I regret including "JSON" in the name of this file :)

Comment (pieh) [reply packages/gatsby/cache-dir/root.js]: Great idea, will also probably rename file as well

Comment (pieh) [reply packages/gatsby/cache-dir/root.js]: Renamed `json-store.js` to `query-result-store.js` + applied component name change suggestions

**Comment by pieh** [comment]
> Let's add a test for our production runtime as well?

Production is not affected by this change (it already works, as we don't use "json-store"s in production), but that's good idea for test parity and potential future regressions. Will add test there as well

**Comment by sidharthachatterjee** [comment]
Yup, I figured we could add one just to ensure we don't see regressions with this behaviour. 

Thanks!

**Comment by oorestisime** [comment]
Thanks you folks! this helps me eliminate some hacky code in my app :)

**Pull request merged**